src/components/CurrencyLogo/index.tsx

```diff
@@ -87,7 +88,17 @@ export default function CurrencyLogo({
   }, [currency, uriLocations])
 
   if (currency?.isNative) {
-    return <StyledEthereumLogo src={EthereumLogo} alt="ethereum logo" size={size} style={style} {...rest} />
+    let nativeLogoUrl: string
+    switch (currency.chainId) {
```

Comment:
have you considered putting this in `CHAIN_INFO` instead? Would probably need to lazy load the logos but might be cleaner

Reply:
not great to import a file from the constants directory, e.g. for your refactoring. i think the logo is a rendering concern

Reply:
just realized we're already doing it for the network logo